crates/fluvio-version-manager/src/common/settings.rs

```diff
@@ -0,0 +1,142 @@
+pub struct Settings {
+    /// The active `channel` for the Fluvio Installation
+    pub channel: Option<Channel>,
+    /// The specific version in use
+    pub version: Option<Version>,
+}
+
+impl Settings {
+    /// Used to create an empty `settings.toml` file. This is used when the user
+    /// installs FVM but no version is set yet.
+    pub fn init() -> Result<Self> {
+        let settings_path = Self::settings_file_path()?;
+
+        if settings_path.exists() {
+            return Err(Error::msg(format!(
+                "Settings already exists on {}",
+                settings_path.display()
+            )));
+        }
+
+        let initial = Self {
+            channel: None,
+            version: None,
+        };
+
+        initial.save()?;
+        tracing::debug!(?settings_path, "Created settings file with success");
+
+        Ok(initial)
+    }
+
+    /// Saves the `settings.toml` file to disk, overwriting the previous version
+    fn save(&self) -> Result<()> {
+        let settings_path = Self::settings_file_path()?;
+        let settings_str = toml::to_string(&self)?;
+
+        write(settings_path, settings_str)?;
+
+        Ok(())
+    }
+
+    /// Retrieves the path to the `settings.toml` file for this host
+    fn settings_file_path() -> Result<PathBuf> {
+        let fvm_path = fvm_workdir_path()?;
+        let settings_path = fvm_path.join(SETTINGS_TOML_FILENAME);
+
+        Ok(settings_path)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::fs::{remove_file, read_to_string, create_dir, remove_dir_all};
+
+    use crate::common::home_dir;
+
+    use super::*;
+
+    fn create_fvm_dir() {
+        let fvm_dir = fvm_workdir_path().unwrap();
+
+        if !fvm_dir.exists() {
+            create_dir(&fvm_dir).unwrap();
+        }
+    }
+
+    fn delete_fvm_dir() {
+        let fvm_dir = fvm_workdir_path().unwrap();
+
+        if fvm_dir.exists() {
+            remove_dir_all(&fvm_dir).unwrap();
+        }
+    }
+
+    #[test]
+    fn test_settings_file_path() {
+        let settings_path =
+            Settings::settings_file_path().expect("Failed to get settings.toml path");
+        let home = home_dir().expect("Failed to get home directory");
+
+        assert!(settings_path.is_absolute());
+        assert!(settings_path.starts_with(home));
+        assert!(settings_path.ends_with(SETTINGS_TOML_FILENAME));
+    }
+
+    #[test]
+    fn creates_settings_file() {
+        create_fvm_dir();
+
+        let _settings = Settings::init().expect("Failed to create settings.toml file");
+        let settings_path =
+            Settings::settings_file_path().expect("Failed to get settings.toml path");
+
+        assert!(
+            settings_path.exists(),
+            "the settings file should exist at this point"
+        );
+
+        remove_file(settings_path).expect("Failed to remove settings.toml file");
+        delete_fvm_dir();
+    }
+
+    #[test]
+    fn update_settings_file() {
+        create_fvm_dir();
```

Comment:
why this is calling this? shouldn't installation step should take care of it?

Reply:
This is testing tho Settings file creation only, these ale unit tests on the Settings struct

Reply:
> shouldn't installation step should take care of it?
Yeah! The command takes care of this but the `Settings` struct is not responsible of creating the directory.

In this context we just create the `settings.toml` file, se I have to prepare the environment for unit tests in this context